sensu/metric/prometheus.go

```diff
@@ -0,0 +1,116 @@
+package metric
+
+import (
+	"io"
+	"math"
+	"strings"
+	"time"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+)
+
+const (
+	sensuPromHelpTagName = "prom_help"
+	sensuPromTypeTagName = "prom_type"
+)
+
+// Points
+type Points []*corev2.MetricPoint
+
+// ToProm writes Points to a buffer using the prometheus text exposition format
+func (m Points) ToProm(writer io.Writer) error {
+	metricFamilies := make(map[string]*dto.MetricFamily)
+	for _, point := range m {
+		var family *dto.MetricFamily
+		family, ok := metricFamilies[point.Name]
+		if !ok {
+			name := point.Name
+			var help string
+			metricType := dto.MetricType_UNTYPED
+			for _, tag := range point.Tags {
+				if tag.Name == sensuPromHelpTagName {
+					help = tag.Value
+					continue
+				}
+				if tag.Name == sensuPromTypeTagName {
+					val := strings.ToLower(tag.Value)
+					switch val {
+					case "counter":
```

Comment:
While the checks we wrote only used counter and gauge metrics we probably want to support the other formats supported by Prometheus?

https://github.com/prometheus/client_model/blob/a863571f36499140c0dbeb5d704144490c6f59e6/go/metrics.pb.go#L27-L31

Reply:
Good Q! For some reason last year I believed this wasn't an option, but revisiting today it is unclear to me why I had thought that.

Histograms and summaries are weird, but perhaps could be implemented. Taking a stab at it now.

Reply:
@fguimond  Should handle all prometheus metric types now.